test/extended/cpu_partitioning/crio.go

```diff
@@ -0,0 +1,314 @@
+		}
+		if !found {
+			return nil, fmt.Errorf("expected spec.reserved to be found in PerformanceProfile(%s)", profile.GetName())
+		}
+
+		// Single Node may include both worker and master labels for node selection,
+		// we match on Master first to account for that, since Master get's preference when both
+		// labels are present.
+		if _, ok := selector[nodeMasterLabel]; ok {
+			nodeRoleCPUSet[nodeMasterLabel] = reservedCPUSet
+		} else if _, ok := selector[nodeWorkerLabel]; ok {
+			nodeRoleCPUSet[nodeWorkerLabel] = reservedCPUSet
+		}
+	}
+
+	return nodeRoleCPUSet, nil
+}
+
+// Execute collection script on machine-config-daemon of every node. Marshal the results to get a list of containers
+// running in the cluster.
+func collectContainerInfo(cfg *rest.Config, c kubernetes.Interface, node corev1.Node) ([]crioContainerData, error) {
+
+	listOptions := metav1.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}).String(),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}).String(),
+	}
+
+	pods, err := c.CoreV1().Pods(namespaceMachineConfigOperator).List(context.TODO(), listOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(pods.Items) < 1 {
+		return nil, fmt.Errorf("failed to get machine-config-daemon pod for the node %q", node.Name)
+	}
+
+	pod := &pods.Items[0]
+
+	var out []byte
+	getInfo := []string{
+		"chroot",
+		"/rootfs",
+		"/bin/bash",
+		"-c", collectionScript}
+
+	ctx := context.TODO()
+	if err := wait.PollWithContext(ctx, 15*time.Second, time.Minute, func(context.Context) (done bool, err error) {
+		out, err = execCommandOnPod(cfg, c, pod, getInfo)
+		if err != nil {
+			return false, err
+		}
+
+		return len(out) != 0, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	info := []crioContainerData{}
+	if err := json.Unmarshal(out, &info); err != nil {
+		return nil, fmt.Errorf("error parsing container output to json: %w", err)
+	}
+
+	return info, nil
+}
+
+func execCommandOnPod(cfg *rest.Config, c kubernetes.Interface, pod *corev1.Pod, command []string) ([]byte, error) {
+	var outputBuf bytes.Buffer
+	var errorBuf bytes.Buffer
+
+	req := c.CoreV1().RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: pod.Spec.Containers[0].Name,
+			Command:   command,
+			Stdout:    true,
+			Stderr:    true,
+		}, scheme.ParameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(cfg, "POST", req.URL())
+	if err != nil {
+		return nil, err
+	}
+
+	err = exec.StreamWithContext(context.Background(), remotecommand.StreamOptions{
+		Stdout: &outputBuf,
+		Stderr: &errorBuf,
+		Tty:    false,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to run command %v: output %q; error %q; %w", command, outputBuf.String(), errorBuf.String(), err)
+	}
+
+	if errorBuf.Len() != 0 {
```

Comment:
for a general utility having messages on stderr is not necessarily a failure conditions - logs for example are usually (and better) sent on stderr

Reply:
That's fair, for this purpose though, in this exec we're just executing a script so I wouldn't expect anything to come through on stderr. If something did, like a log, then I think we would consider that a problem since it's unexpected for this call